tests/integration/high_availability/test_self_healing.py

```diff
@@ -222,20 +212,29 @@ async def test_network_cut(ops_test: OpsTest):
     # verify that connection is possible
     assert is_connection_possible(config), "❌ Connection is not possible after network restore"
 
-    # verify instance is part of the cluster
-    logger.info("Check if instance in cluster")
-    assert await is_unit_in_cluster(
-        ops_test, primary_unit.name, another_unit.name
-    ), "Unit not online in the cluster"
+    # ensure continuous writes still incrementing for all units
+    async with ops_test.fast_forward():
+        # allow extra time for relation data to be updated, accommodating CI lag
+        sleep(60)
```

Comment:
If this is giving you trouble, then maybe there is a condition that you could check and then you could throw in an `ops.model.wait_for`